tools/websocket_to_posix_proxy/src/websocket_to_posix_proxy.c

```diff
@@ -1438,7 +1438,7 @@ void Getaddrinfo(int client_fd, uint8_t *data, uint64_t numBytes) {
 
   struct addrinfo *res = 0;
   int ret = getaddrinfo(d->node, d->service, d->hasHints ? &hints : 0, &res);
-  int errorCode = (ret != 0) ? GET_SOCKET_ERROR() : 0;
+  int errorCode = SOCKET_API_CALL_RETURNS_AN_ERROR(ret) ? GET_SOCKET_ERROR() : 0;
```

Comment:
This line being repeated so often makes me thing perhaps a highter level macro would be useful (e.g. `HANDLE_SOCKET_API_ERROR`?

Reply:
Good idea, created a dedicated function for this.